File: proposed-projects/round-trip-experiment/reverse/20260909T002206Z_273cf6b9/20260910T042804Z_e72f6c04/ported_implementation/gbnf/grammar_graph/rule_ref_test.py
import pytest

from .rule_ref import RuleRef


def test_initializes_with_a_given_value():
    rule_ref = RuleRef(123)
    assert rule_ref.value == 123


def test_allows_setting_and_getting_nodes():
    mock_nodes = [1]
    rule_ref = RuleRef(123)
    rule_ref.nodes = mock_nodes
    assert rule_ref.nodes == mock_nodes


def test_raises_an_error_if_trying_to_get_nodes_before_setting():
    rule_ref = RuleRef(123)
    with pytest.raises(ValueError, match="Nodes are not set"):
        rule_ref.nodes

File: packages/gbnf-experiment/data/20260909T124157Z_4ee67ba2/ported_implementation/tests/unit/grammar-graph/rule-ref.test.ts
import assert from 'node:assert/strict';
import { describe, test } from 'node:test';
import { RuleRef } from '../../../src/grammar-graph/rule-ref.ts';
import type { GraphNode } from '../../../src/grammar-graph/graph-node.ts';

describe('rule ref', () => {
  test('initializes with a given value', () => {
    assert.equal(new RuleRef(123).value, 123);
  });

  test('allows setting and getting nodes', () => {
    const mockNodes = new Set([1 as unknown as GraphNode]);
    const ruleRef = new RuleRef(123);
    ruleRef.nodes = mockNodes;
    assert.deepStrictEqual(ruleRef.nodes, mockNodes);
  });

  test('throws an error if trying to get nodes before setting', () => {
    const ruleRef = new RuleRef(123);
    assert.throws(() => ruleRef.nodes, { message: 'Nodes are not set' });
  });
});
